include/mgp_mock.py

```diff
@@ -0,0 +1,1011 @@
+
+        for item in self.items():
+            yield item.value
+
+    def __len__(self) -> int:
+        if not self._vertex_or_edge.is_valid():
+            raise InvalidContextError()
+
+        if self._len is None:
+            self._len = sum(1 for _ in self.items())
+
+        return self._len
+
+    def __iter__(self) -> typing.Iterable[str]:
+        if not self._vertex_or_edge.is_valid():
+            raise InvalidContextError()
+
+        for item in self.items():
+            yield item.name
+
+    def __getitem__(self, property_name: str) -> object:
+        if not self._vertex_or_edge.is_valid():
+            raise InvalidContextError()
+
+        vertex_or_edge_props = (
+            self._vertex_or_edge.graph.nodes[self._vertex_or_edge.id]
+            if isinstance(self._vertex_or_edge, _mgp_mock.Vertex)
+            else self._vertex_or_edge.graph.edges[self._vertex_or_edge.edge]
+        )
+
+        return vertex_or_edge_props[property_name]
+
+    def __setitem__(self, property_name: str, value: object) -> None:
+        if not self._vertex_or_edge.is_valid():
+            raise InvalidContextError()
+
+        if nx.is_frozen(self._vertex_or_edge.graph):
+            raise ImmutableObjectError("Cannot modify immutable object.")
+
+        if isinstance(self._vertex_or_edge, _mgp_mock.Vertex):
+            vertex_id = self._vertex_or_edge.id
+            self._vertex_or_edge.graph.nodes[vertex_id][property_name] = value
+            return
+
+        edge_ids = self._vertex_or_edge.edge
+        self._vertex_or_edge.graph.edges[edge_ids][property_name] = value
+
+    def __contains__(self, property_name: str) -> bool:
+        if not self._vertex_or_edge.is_valid():
+            raise InvalidContextError()
+
+        try:
+            _ = self[property_name]
+            return True
+        except KeyError:
+            return False
+
+
+class EdgeType:
+    __slots__ = ("_name",)
+
+    def __init__(self, name):
+        self._name = name
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    def __eq__(self, other) -> bool:
+        if isinstance(other, EdgeType):
+            return self.name == other.name
+
+        if isinstance(other, str):
+            return self.name == other
+
+        return NotImplemented
+
+
+if sys.version_info >= (3, 5, 2):
+    EdgeId = typing.NewType("EdgeId", int)
+else:
+    EdgeId = int
+
+
+class Edge:
+    __slots__ = ("_edge",)
+
+    def __init__(self, edge):
+        if not isinstance(edge, _mgp_mock.Edge):
+            raise TypeError(f"Expected '_mgp_mock.Edge', got '{type(edge)}'")
+
+        self._edge = edge
+
+    def __deepcopy__(self, memo):
+        # This is the same as the shallow copy, because we want to share the
+        # underlying C struct. Besides, it doesn't make much sense to actually
+        # copy _mgp_mock.Edge as that is actually a reference to a graph element
+        # and not a proper value.
```

Comment:
Is this comment here necessary, we are using mock here?

Reply:
Well, if the Python API implements shallow copying in place of deep copying, I’ll add the same so that users don’t experience unexpected behavior upon switching from the mock to the real API.